android/src/main/java/com/tailscale/ipn/ui/view/TailnetLockSetupView.kt

```diff
@@ -0,0 +1,118 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.tailscale.ipn.R
+import com.tailscale.ipn.ui.Links
+import com.tailscale.ipn.ui.theme.ts_color_light_blue
+import com.tailscale.ipn.ui.util.ClipboardValueView
+import com.tailscale.ipn.ui.util.LoadingIndicator
+import com.tailscale.ipn.ui.viewModel.TailnetLockSetupViewModel
+import com.tailscale.ipn.ui.viewModel.TailnetLockSetupViewModelFactory
+
+@Composable
+fun TailnetLockSetupView(
+    nav: BackNavigation,
+    model: TailnetLockSetupViewModel = viewModel(factory = TailnetLockSetupViewModelFactory())
+) {
+  val statusItems = model.statusItems.collectAsState().value
+  val nodeKey = model.nodeKey.collectAsState().value
+  val tailnetLockKey = model.tailnetLockKey.collectAsState().value
+
+  Scaffold(topBar = { Header(R.string.tailnet_lock, onBack = nav.onBack) }) { innerPadding ->
+    LoadingIndicator.Wrap {
+      LazyColumn(
+          verticalArrangement = Arrangement.spacedBy(10.dp),
+          modifier = Modifier.padding(innerPadding)) {
+            item(key = "header") {
+              ExplainerView()
+              Spacer(Modifier.size(4.dp))
+            }
+
+            items(items = statusItems) { statusItem ->
+              Row(
+                  horizontalArrangement = Arrangement.spacedBy(12.dp),
+                  modifier = Modifier.padding(horizontal = 16.dp)) {
+                    Icon(
+                        painter = painterResource(id = statusItem.icon),
+                        contentDescription = null,
+                        tint = ts_color_light_blue)
+                    Text(stringResource(statusItem.title))
+                  }
+            }
+
+            item(key = "nodeKey") {
+              Column(
+                  verticalArrangement = Arrangement.spacedBy(8.dp),
+                  modifier = Modifier.padding(horizontal = 16.dp)) {
+                    Spacer(Modifier.size(4.dp))
+                    ClipboardValueView(
+                        value = nodeKey,
```

Comment:
this can be null right? since we do nodeKey.set(result.getOrNull()?.NodeKey)
should this check if null and have a fallback value if null?

Reply:
Fixed